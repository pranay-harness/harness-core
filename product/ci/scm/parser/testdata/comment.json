{
  "action": "created",
  "issue": {
    "url": "https://api.github.com/repos/abc/def-ci-webhook-test/issues/2",
    "repository_url": "https://api.github.com/repos/abc/def-ci-webhook-test",
    "labels_url": "https://api.github.com/repos/abc/def-ci-webhook-test/issues/2/labels{/name}",
    "comments_url": "https://api.github.com/repos/abc/def-ci-webhook-test/issues/2/comments",
    "events_url": "https://api.github.com/repos/abc/def-ci-webhook-test/issues/2/events",
    "html_url": "https://github.com/abc/def-ci-webhook-test/pull/2",
    "id": 735678167,
    "node_id": "MDExOlB1bGxSZXF1ZXN0NTE1MDI2Njgw",
    "number": 2,
    "title": "Update README.md",
    "user": {
      "login": "lts-def",
      "id": 10278482,
      "node_id": "MDQ6VXNlcjEwMjc4NDgy",
      "avatar_url": "https://avatars0.githubusercontent.com/u/10278482?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/lts-def",
      "html_url": "https://github.com/lts-def",
      "followers_url": "https://api.github.com/users/lts-def/followers",
      "following_url": "https://api.github.com/users/lts-def/following{/other_user}",
      "gists_url": "https://api.github.com/users/lts-def/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/lts-def/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/lts-def/subscriptions",
      "organizations_url": "https://api.github.com/users/lts-def/orgs",
      "repos_url": "https://api.github.com/users/lts-def/repos",
      "events_url": "https://api.github.com/users/lts-def/events{/privacy}",
      "received_events_url": "https://api.github.com/users/lts-def/received_events",
      "type": "User",
      "site_admin": false
    },
    "labels": [

    ],
    "state": "open",
    "locked": false,
    "assignee": null,
    "assignees": [

    ],
    "milestone": null,
    "comments": 4,
    "created_at": "2020-11-03T22:32:14Z",
    "updated_at": "2020-12-29T05:49:14Z",
    "closed_at": null,
    "author_association": "COLLABORATOR",
    "active_lock_reason": null,
    "pull_request": {
      "url": "https://api.github.com/repos/abc/def-ci-webhook-test/pulls/2",
      "html_url": "https://github.com/abc/def-ci-webhook-test/pull/2",
      "diff_url": "https://github.com/abc/def-ci-webhook-test/pull/2.diff",
      "patch_url": "https://github.com/abc/def-ci-webhook-test/pull/2.patch"
    },
    "body": "",
    "performed_via_github_app": null
  },
  "comment": {
    "url": "https://api.github.com/repos/abc/def-ci-webhook-test/issues/comments/751955664",
    "html_url": "https://github.com/abc/def-ci-webhook-test/pull/2#issuecomment-751955664",
    "issue_url": "https://api.github.com/repos/abc/def-ci-webhook-test/issues/2",
    "id": 751955664,
    "node_id": "MDEyOklzc3VlQ29tbWVudDc1MTk1NTY2NA==",
    "user": {
      "login": "lts-def",
      "id": 10278482,
      "node_id": "MDQ6VXNlcjEwMjc4NDgy",
      "avatar_url": "https://avatars0.githubusercontent.com/u/10278482?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/lts-def",
      "html_url": "https://github.com/lts-def",
      "followers_url": "https://api.github.com/users/lts-def/followers",
      "following_url": "https://api.github.com/users/lts-def/following{/other_user}",
      "gists_url": "https://api.github.com/users/lts-def/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/lts-def/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/lts-def/subscriptions",
      "organizations_url": "https://api.github.com/users/lts-def/orgs",
      "repos_url": "https://api.github.com/users/lts-def/repos",
      "events_url": "https://api.github.com/users/lts-def/events{/privacy}",
      "received_events_url": "https://api.github.com/users/lts-def/received_events",
      "type": "User",
      "site_admin": false
    },
    "created_at": "2020-12-29T05:49:14Z",
    "updated_at": "2020-12-29T05:49:14Z",
    "author_association": "COLLABORATOR",
    "body": "trigger abc",
    "performed_via_github_app": null
  },
  "repository": {
    "id": 309651765,
    "node_id": "MDEwOlJlcG9zaXRvcnkzMDk2NTE3NjU=",
    "name": "def-ci-webhook-test",
    "full_name": "abc/def-ci-webhook-test",
    "private": true,
    "owner": {
      "login": "abc",
      "id": 18273000,
      "node_id": "MDEyOk9yZ2FuaXphdGlvbjE4MjczMDAw",
      "avatar_url": "https://avatars1.githubusercontent.com/u/18273000?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/abc",
      "html_url": "https://github.com/abc",
      "followers_url": "https://api.github.com/users/abc/followers",
      "following_url": "https://api.github.com/users/abc/following{/other_user}",
      "gists_url": "https://api.github.com/users/abc/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/abc/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/abc/subscriptions",
      "organizations_url": "https://api.github.com/users/abc/orgs",
      "repos_url": "https://api.github.com/users/abc/repos",
      "events_url": "https://api.github.com/users/abc/events{/privacy}",
      "received_events_url": "https://api.github.com/users/abc/received_events",
      "type": "Organization",
      "site_admin": false
    },
    "html_url": "https://github.com/abc/def-ci-webhook-test",
    "description": "Webhook test",
    "fork": false,
    "url": "https://api.github.com/repos/abc/def-ci-webhook-test",
    "forks_url": "https://api.github.com/repos/abc/def-ci-webhook-test/forks",
    "keys_url": "https://api.github.com/repos/abc/def-ci-webhook-test/keys{/key_id}",
    "collaborators_url": "https://api.github.com/repos/abc/def-ci-webhook-test/collaborators{/collaborator}",
    "teams_url": "https://api.github.com/repos/abc/def-ci-webhook-test/teams",
    "hooks_url": "https://api.github.com/repos/abc/def-ci-webhook-test/hooks",
    "issue_events_url": "https://api.github.com/repos/abc/def-ci-webhook-test/issues/events{/number}",
    "events_url": "https://api.github.com/repos/abc/def-ci-webhook-test/events",
    "assignees_url": "https://api.github.com/repos/abc/def-ci-webhook-test/assignees{/user}",
    "branches_url": "https://api.github.com/repos/abc/def-ci-webhook-test/branches{/branch}",
    "tags_url": "https://api.github.com/repos/abc/def-ci-webhook-test/tags",
    "blobs_url": "https://api.github.com/repos/abc/def-ci-webhook-test/git/blobs{/sha}",
    "git_tags_url": "https://api.github.com/repos/abc/def-ci-webhook-test/git/tags{/sha}",
    "git_refs_url": "https://api.github.com/repos/abc/def-ci-webhook-test/git/refs{/sha}",
    "trees_url": "https://api.github.com/repos/abc/def-ci-webhook-test/git/trees{/sha}",
    "statuses_url": "https://api.github.com/repos/abc/def-ci-webhook-test/statuses/{sha}",
    "languages_url": "https://api.github.com/repos/abc/def-ci-webhook-test/languages",
    "stargazers_url": "https://api.github.com/repos/abc/def-ci-webhook-test/stargazers",
    "contributors_url": "https://api.github.com/repos/abc/def-ci-webhook-test/contributors",
    "subscribers_url": "https://api.github.com/repos/abc/def-ci-webhook-test/subscribers",
    "subscription_url": "https://api.github.com/repos/abc/def-ci-webhook-test/subscription",
    "commits_url": "https://api.github.com/repos/abc/def-ci-webhook-test/commits{/sha}",
    "git_commits_url": "https://api.github.com/repos/abc/def-ci-webhook-test/git/commits{/sha}",
    "comments_url": "https://api.github.com/repos/abc/def-ci-webhook-test/comments{/number}",
    "issue_comment_url": "https://api.github.com/repos/abc/def-ci-webhook-test/issues/comments{/number}",
    "contents_url": "https://api.github.com/repos/abc/def-ci-webhook-test/contents/{+path}",
    "compare_url": "https://api.github.com/repos/abc/def-ci-webhook-test/compare/{base}...{head}",
    "merges_url": "https://api.github.com/repos/abc/def-ci-webhook-test/merges",
    "archive_url": "https://api.github.com/repos/abc/def-ci-webhook-test/{archive_format}{/ref}",
    "downloads_url": "https://api.github.com/repos/abc/def-ci-webhook-test/downloads",
    "issues_url": "https://api.github.com/repos/abc/def-ci-webhook-test/issues{/number}",
    "pulls_url": "https://api.github.com/repos/abc/def-ci-webhook-test/pulls{/number}",
    "milestones_url": "https://api.github.com/repos/abc/def-ci-webhook-test/milestones{/number}",
    "notifications_url": "https://api.github.com/repos/abc/def-ci-webhook-test/notifications{?since,all,participating}",
    "labels_url": "https://api.github.com/repos/abc/def-ci-webhook-test/labels{/name}",
    "releases_url": "https://api.github.com/repos/abc/def-ci-webhook-test/releases{/id}",
    "deployments_url": "https://api.github.com/repos/abc/def-ci-webhook-test/deployments",
    "created_at": "2020-11-03T10:36:21Z",
    "updated_at": "2020-12-29T05:06:28Z",
    "pushed_at": "2020-12-29T05:06:25Z",
    "git_url": "git://github.com/abc/def-ci-webhook-test.git",
    "ssh_url": "git@github.com:abc/def-ci-webhook-test.git",
    "clone_url": "https://github.com/abc/def-ci-webhook-test.git",
    "svn_url": "https://github.com/abc/def-ci-webhook-test",
    "homepage": null,
    "size": 338096,
    "stargazers_count": 0,
    "watchers_count": 0,
    "language": "Java",
    "has_issues": true,
    "has_projects": true,
    "has_downloads": true,
    "has_wiki": true,
    "has_pages": false,
    "forks_count": 0,
    "mirror_url": null,
    "archived": false,
    "disabled": false,
    "open_issues_count": 2,
    "license": null,
    "forks": 0,
    "open_issues": 2,
    "watchers": 0,
    "default_branch": "master"
  },
  "organization": {
    "login": "abc",
    "id": 18273000,
    "node_id": "MDEyOk9yZ2FuaXphdGlvbjE4MjczMDAw",
    "url": "https://api.github.com/orgs/abc",
    "repos_url": "https://api.github.com/orgs/abc/repos",
    "events_url": "https://api.github.com/orgs/abc/events",
    "hooks_url": "https://api.github.com/orgs/abc/hooks",
    "issues_url": "https://api.github.com/orgs/abc/issues",
    "members_url": "https://api.github.com/orgs/abc/members{/member}",
    "public_members_url": "https://api.github.com/orgs/abc/public_members{/member}",
    "avatar_url": "https://avatars1.githubusercontent.com/u/18273000?v=4",
    "description": ""
  },
  "sender": {
    "login": "lts-def",
    "id": 10278482,
    "node_id": "MDQ6VXNlcjEwMjc4NDgy",
    "avatar_url": "https://avatars0.githubusercontent.com/u/10278482?v=4",
    "gravatar_id": "",
    "url": "https://api.github.com/users/lts-def",
    "html_url": "https://github.com/lts-def",
    "followers_url": "https://api.github.com/users/lts-def/followers",
    "following_url": "https://api.github.com/users/lts-def/following{/other_user}",
    "gists_url": "https://api.github.com/users/lts-def/gists{/gist_id}",
    "starred_url": "https://api.github.com/users/lts-def/starred{/owner}{/repo}",
    "subscriptions_url": "https://api.github.com/users/lts-def/subscriptions",
    "organizations_url": "https://api.github.com/users/lts-def/orgs",
    "repos_url": "https://api.github.com/users/lts-def/repos",
    "events_url": "https://api.github.com/users/lts-def/events{/privacy}",
    "received_events_url": "https://api.github.com/users/lts-def/received_events",
    "type": "User",
    "site_admin": false
  }
}