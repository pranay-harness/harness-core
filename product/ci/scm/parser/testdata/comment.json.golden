{
   "comment":{
      "action":"CREATE",
      "repo":{
         "id":"309651765",
         "namespace":"abc",
         "name":"def-ci-webhook-test",
         "branch":"master",
         "private":true,
         "clone":"https://github.com/abc/def-ci-webhook-test.git",
         "cloneSsh":"git@github.com:abc/def-ci-webhook-test.git",
         "link":"https://github.com/abc/def-ci-webhook-test",
         "created":"0001-01-01T00:00:00Z",
         "updated":"0001-01-01T00:00:00Z"
      },
      "issue":{
         "number":2,
         "title":"Update README.md",
         "link":"https://github.com/abc/def-ci-webhook-test/pull/2",
         "user":{
            "login":"lts-def",
            "avatar":"https://avatars0.githubusercontent.com/u/10278482?v=4",
            "created":"0001-01-01T00:00:00Z",
            "updated":"0001-01-01T00:00:00Z"
         },
         "pr":{
            "link":"https://github.com/abc/def-ci-webhook-test/pull/2",
            "base":{

            },
            "head":{

            },
            "author":{
               "created":"0001-01-01T00:00:00Z",
               "updated":"0001-01-01T00:00:00Z"
            },
            "created":"0001-01-01T00:00:00Z",
            "updated":"0001-01-01T00:00:00Z"
         },
         "created":"2020-11-03T22:32:14Z",
         "updated":"2020-12-29T05:49:14Z"
      },
      "sender":{
         "login":"lts-def",
         "avatar":"https://avatars0.githubusercontent.com/u/10278482?v=4",
         "created":"0001-01-01T00:00:00Z",
         "updated":"0001-01-01T00:00:00Z"
      },
      "comment":{
         "id":751955664,
         "body":"trigger abc",
         "user":{
            "login":"lts-def",
            "avatar":"https://avatars0.githubusercontent.com/u/10278482?v=4",
            "created":"0001-01-01T00:00:00Z",
            "updated":"0001-01-01T00:00:00Z"
         },
         "created":"2020-12-29T05:49:14Z",
         "updated":"2020-12-29T05:49:14Z"
      }
   }
}